test/numbers.jl

```diff
@@ -2867,14 +2867,14 @@ end
 
 @testset "iszero" begin
     # Numeric scalars
-    for T in Iterators.flatten(subtypes.([AbstractFloat, Signed, Unsigned]))
+    for T in [Float16, Float32, Float64, BigFloat,
+              Int8, Int16, Int32, Int64, Int128, BigInt,
+              UInt8, UInt16, UInt32, UInt64, UInt128,
+              Rational]
```

Comment:
This makes me realize that we could use `iszero(x::Rational) = iszero(x.num)`, which saves a comparison compared to the fallback method that does `== 0`